### ISSUE TITLE
CallLogger fails to intercept exceptions

### ISSUE THREAD

**Issue by kyegupov** [Kotlin/ktor#50]
ktor 0.2.2
CallLogger fails to properly capture exceptions (NPE, for example) and logs them instead as "trace"s.

PipelineMachine ends up in `PipelineState.Succeeded` state. My guess is StatusPagesKt.errorPage creates a new execution with state "Executing", which clears the "Failed" state from the pipeline".

As a side note, that whole "Pipeline" stuff is absolutely incomprehensible and hard to debug. Took me ah hour to understand it's actually an implementation of event loop. I have a bad feeling about this approach (reminds of JavaScript callback hell - highly sharded code hard to follow, complex state management), but at the very least it needs tons, TONS of documentation.

**Comment by orangy** [comment]
Please check what is in master branch, it has been cleared a lot. Also, several "missing exceptions" cases were fixed. We are going to release 0.2.3 pretty soon, we are working on a decision with one API detail about ApplicationModules.

**Comment by kyegupov** [comment]
Maybe it's worth to add a warning to README.md notifying potential users about instability?
ktor, being developed by JB, is a "default" choice for a Kotlin web fw, and in post-Node.js era people are quite keen on using versions 0.x, but at the moment the amount of gotchas and the learning curve is going to frustrate many people who try naively to use 0.2.2

**Comment by orangy** [comment]
I think this has been fixed.
